RadixWallet/Clients/AccountLockersClient/AccountLockersClient+Live.swift

```diff
@@ -0,0 +1,210 @@
+					for lockerState in lockerStates.items {
+						if let cached = getCachedVersionIfValid(lockerState: lockerState) {
+							// The cached claim information is valid.
+							accountLockerClaims.append(cached)
+
+						} else {
+							// The cache is outdated or doesn't exist, so we will fetch its details from Gateway
+							let lockerAddress = lockerState.lockerAddress
+							let accountAddress = lockerState.accountAddress
+							let lockerContent = try await getLockerContent(lockerAddress: lockerAddress, accountAddress: accountAddress)
+							let claims = lockerContent.filter(\.isValidClaim)
+
+							guard let dapp = dappsWithLockers.first(where: { $0.lockerAddress == lockerAddress })?.dapp else {
+								fatalError("Programmer error: there should be a dapp for the given locker")
+							}
+
+							let accountLockerClaim = AccountLockerClaims(
+								lockerAddress: lockerAddress,
+								accountAddress: accountAddress,
+								dappDefinitionAddress: dapp.dappDefinitionAddress.address,
+								dappName: dapp.displayName,
+								lastTouchedAtStateVersion: lockerState.lastTouchedAtStateVersion,
+								claims: claims
+							)
+
+							// Cache the result and append to list
+							cacheClient.save(accountLockerClaim, .accountLockerClaims(accountAddress: accountAddress, lockerAddress: lockerAddress))
+							accountLockerClaims.append(accountLockerClaim)
+						}
+					}
+
+					// Set the claims for the given account, filtering those that are empty.
+					claimsPerAccount[account.address] = accountLockerClaims.filter(not(\.claims.isEmpty))
+				}
+
+				// Emit a new result
+				claimsPerAccountSubject.send(claimsPerAccount)
+			}
+		}
+
+		@Sendable
+		func filterDappsWithAccountLockers(_ dapps: AuthorizedDapps) async throws -> [DappWithLockerAddress] {
+			// Filter dapps for which user doesn't want to see deposits
+			let dappsWithDepositsVisible = dapps
+				.filter(\.isDepositsVisible)
+				.map(\.dappDefinitionAddress.asGeneral)
+
+			// Get entity details from those dapps, to filter only those that actually have a primary locker.
+			let dappsWithLocker = try await onLedgerEntitiesClient
+				.getEntities(dappsWithDepositsVisible, .init(dappTwoWayLinks: true), nil, .useCache, false)
+				.compactMap(\.account)
+				.compactMap { account -> (AccountAddress, String)? in
+					guard let lockerAddress = account.details?.primaryLocker else {
+						return nil
+					}
+					return (account.address, lockerAddress)
+				}
+
+			// Loop over the original input to only return those that actually have a primary locker
+			return dapps.compactMap { dapp in
+				guard let dappWithLocker = dappsWithLocker.first(where: { $0.0 == dapp.dappDefinitionAddress }) else {
+					return nil
+				}
+				return DappWithLockerAddress(dapp: dapp, lockerAddress: dappWithLocker.1)
+			}
+		}
+
+		@Sendable
+		func getLockersPerAccount(accounts: Accounts, dapps: [DappWithLockerAddress]) -> [AccountAddress: [String]] {
```

Comment:
Why String? Should be `LockerAddress` right? :)

Reply:
well, to use `LockerAddress` we would need to use the throwing init that takes a Strings and validates it.

Given we will later use its String value only, I don't see the point in doing that actually.

Reply:
ended up adding every address type: `LockerAddress`, `ResourceAddress`, `AccountAddress` & `VaultAddress`

Reply:
where are we using it as string later? we should do it as late as possible. Yes there is meaning in going through the address type, **like we do everywhere else in the app**, and that is validating that it is a valid address. that is always beneficial.

Reply:
What I meant is that the `lockerAddress` that is fetched here as a String, is only [used](https://github.com/radixdlt/babylon-wallet-ios/pull/1312/files#diff-8072040d0aad8663b6de7c4bf866234dbe8e39c95d98f94922aab09c16e9e20cR158) to build the `GatewayAPI.AccountLockerAddress` that takes a String as well. Will update so that we parse it as a `LockerAddress` and then get its `var address: String` when needed.